### ISSUE TITLE
Deserialization issue on mono/OS X

### ISSUE THREAD

**Issue by marcuslindblom** [aspnet/Mvc#2754]
Where can I specify the assembly name when using mono/OS X? I experience a deserialization problem and I think it's because the assembly name may be different from when I use VS 2015?

**Comment by davidfowl** [comment]
Can you give some more details:

- A stack trace of the error
- An example that shows the problem posted to a github repository

**Comment by marcuslindblom** [comment]
This is the [actual error](http://f.cl.ly/items/1e1Y1w3R0P193V1q3e0v/Internal%20Server%20Error.html) I can write a small example if that helps.

**Comment by davidfowl** [comment]
Yes, please write a minimal example. It also doesn't look specific to OSX. It also looks like there's use of dynamic somewhere in the example.

**Comment by marcuslindblom** [comment]
Here is the sample code you can use to reproduce the error. https://github.com/marcuslindblom/test, You need to download RavenDB to run the code because Raven can not run in embedded mode on OS X.

The exception:
```
NullReferenceException: Object reference not set to an instance of an object

Microsoft.Framework.Runtime.ProjectExportProviderHelper.GetExportsRecursive (ICache cache, ILibraryManager manager, ILibraryExportProvider libraryExportProvider, ILibraryKey target, System.Func`2 include) [0x00000] in <filename unknown>, line 0
```

**Comment by marcuslindblom** [comment]
Actually you need to run RavenDB on a windows machine, sorry about that but I didn't now how to reproduce it otherwise

**Comment by danroth27** [comment]
@sornaks 

I don't see any deserialization in the stack trace. Looks like an issue invoking the action method using reflection.

**Comment by rynowak** [comment]
@marcuslindblom - sorry for the delay in getting to this.

Looking at the error message that you linked, it seems to tell you everything you need to know: ` Object type Raven.Abstractions.Linq.DynamicJsonObject cannot be converted to target type: BrickPile.Models.Home`

There's no feature in MVC that will convert an arbitrary `dynamic` object into your model type.

Taking a look at your sample project now.

**Comment by rynowak** [comment]
I tried your sample in the linked repo, and works for me on windows without issue.

**Comment by marcuslindblom** [comment]
When you say, "work as expected in your sample", do you mean that using dynamic should work on Windows but not on OS X/Mono? The reason I'm using dynamic is because I build a framework/cms and the type is unknown to me. Using the exact type will work on both systems.

**Comment by rynowak** [comment]
> work as expected in your sample.

I mean that your provided sample code with `response.WriteAsync(data.Title)` correctly retrieves the title from the database and writes it.

> The reason I'm using dynamic is because I build a framework/cms and the type is unknown to me. Using the exact type will work on both systems.

The error message that you included shows trying to call an action method with an `Raven.Abstractions.Linq.DynamicJsonObject` in action arguments dictionary instead of an `BrickPile.Models.Home`. If you're writing a model binder that's retrieving this data from the DB, then you'll need to convert it to the appropriate type, which should be available on the corresponding parameter descriptor.

**Comment by marcuslindblom** [comment]
@rynowak I found the problem with my code and maybe it is a RavenDB problem, I can't say, but when I save a new object to RavenDB from OS X the `Raven-Clr-Type` for the new object differs from when I save it from my Windows machine using Visual Studio 2015. On my Windows machine the assembly name is the one I have set on the project level.

<img width="394" alt="screen shot 2015-07-20 at 20 57 55" src="https://cloud.githubusercontent.com/assets/319720/8784391/27e58bd4-2f22-11e5-86b9-318a87225931.png">

When I create and save a `new BrickPile.Models.Home()` on my Windows machine to RavenDB the `Raven-Clr-Type` looks like this:

<img width="273" alt="screen shot 2015-07-20 at 20 56 38" src="https://cloud.githubusercontent.com/assets/319720/8784456/864725e8-2f22-11e5-9f7b-2d0c76a4f4d2.png">

When I create and save a `new BrickPile.Models.Home()` on my OS X machine using Visual Studio Code to RavenDB the `Raven-Clr-Type` looks like this:

<img width="285" alt="screen shot 2015-07-20 at 21 03 00" src="https://cloud.githubusercontent.com/assets/319720/8784510/c5ed68e2-2f22-11e5-8e10-77a74e82e273.png">

So the OS X machine needs the assembly name to be `src` and the Windows Machine uses `brics` which is the name I gave my project. Maybe it's possible to set the name in `global.json` because OS X/Code seems to use the folder name as the assembly name.

My `global.json` is the same on both machines and looks like this.

```json
{
    "sources": [ "src", "test" ],
    "sdk": {
        "version": "1.0.0-beta5"
    }
}
```
